File: src/cpp_pkg/src/server_node.cpp
// Based on the AddTwoInts interface
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using namespace std::placeholders;

class ServerNode : public rclcpp::Node
{
    public:
        ServerNode() : Node("server_node")
        {
            server_ = this->create_service<example_interfaces::srv::AddTwoInts>("add_two_ints", std::bind(&ServerNode::callback_server, this, _1, _2));

            RCLCPP_INFO(this->get_logger(), "Server node is running ...");
        }
        
    private:
        void callback_server(const example_interfaces::srv::AddTwoInts::Request::SharedPtr req, const example_interfaces::srv::AddTwoInts::Response::SharedPtr res)
        {
            res->sum = req->a + req->b;
        }

        rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;
}; 

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<ServerNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/py_pkg/py_pkg/server_node.py
#!/usr/bin/env python3
# Based on the AddTwoInts interface
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


class ServerNode(Node):

    def __init__(self):
        super().__init__("server_node")
        self.server_ = self.create_service(AddTwoInts, "add_two_ints", self.callback_server)
        self.get_logger().info("Server node is running ...")

    def callback_server(self, req: AddTwoInts.Request, res: AddTwoInts.Response):
        res.sum = req.a + req.b
        return res

def main(args=None):
    rclpy.init(args=args)
    node = ServerNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()